app/services/browser_support.rb

```diff
@@ -0,0 +1,10 @@
+class BrowserSupport
+  def self.supported?(user_agent)
+    matcher = BrowserslistUseragent::Match.new(browser_support_config, user_agent)
+    matcher.browser? && matcher.version?(allow_higher: true)
+  end
+
+  def self.browser_support_config
+    @browser_support_config ||= JSON.parse(File.read(Rails.root.join('browsers.json')))
```

Comment:
Consider how to handle failures such as when the file doesn't exist.

Reply:
Maybe we have the app read the `browsers.json` into a constant in an initializer, so that way it fails very loudly + app doesn't boot if the file doesn't exist?

Reply:
Mm, yeah, I thought we'd done something like that with `AssetSources`, which has a similar [manifest load procedure](https://github.com/18F/identity-idp/blob/4607236ebf449e38c12b01d09ee13f22195d01f5/lib/asset_sources.rb#L40-L46). But in looking closer, all we did was [set a flag in `application.rb`](https://github.com/18F/identity-idp/blob/4607236ebf449e38c12b01d09ee13f22195d01f5/config/application.rb#L40). I could see something similar where we'd preemptively call a `.load_browser_support_config!` in one of those files?

My first thought was to just have some graceful fallback behavior, like either we always/never include it if we're unable to load the configuration for some reason.

Really, the only realistic case I'm worried about here is if a local developer were to run `rails s` locally without the file being present, so I'm not _overly_ concerned about dealing with this scenario.

Reply:
I went for a "gracefully assume supported if failed to load" approach in latest commits.